tests/test_health.py

```diff
@@ -827,3 +827,32 @@ def test_positive_remove_job_file(setup_subscribe_to_cdn_dogfood, ansible_module
     contacted = ansible_module.find(paths="/var/lib/pulp", file_type="file")
     for file in contacted.values()[0]["files"]:
         assert "job" not in file["path"]
+
+
+@pytest.mark.capsule
+def test_positive_check_non_rh_packages(setup_custom_package, ansible_module):
+    """Verify check-non-redhat-repository.
+
+    :id: 2b7a25bb-902c-4a87-bb71-f460dcce655c
+
+    :setup:
+        1. foreman-maintain should be installed.
+        2. Configure custom repository and install custom package
+
+    :steps:
+        1. Run foreman-maintain health check --label non-rh-packages.
+        3. Verify presense of non-RH/custom package installed on system.
+
+    :BZ: 1869865
+
+    :expectedresults: check-non-rh-packages should work.
+
+    :CaseImportance: High
+    """
+    contacted = ansible_module.command(Health.check({"label": "non-rh-packages"}))
+    for result in contacted.values():
+        logger.info(result["stdout"])
+        assert "Found 1 unexpected non Red Hat Package(s) installed!" in result["stdout"]
+        assert "walrus-5.21-1.noarch" in result["stdout"]
```

Comment:
@Gauravtalreja1 will this `walrus` package version will stay same in near future? or should we change this assertion as `assert "walrus" in result["stdout"]`?

Reply:
There is a rare possibility of changing versions for packages under custom repos, but if you insist I can change this assert as you mentioned

Reply:
no need then.